### PR TITLE
[Enhancement] Unified Configuration Center for StreamX

### PR DESCRIPTION
**Pull request by Al-assad**
Signed-off-by: Al-assad yulin.ying@outlook.com

### What problem does this PR solve?
The current distribution of configuration items and constants in StreamX is confusing. All configuration comes from system parameters, spring boot configuration files and static configuration in ConfigConst. However, swapping configuration overrides via System Properties is non-thread-safe, which is not a good practice and does not facilitate subsequent expansion.

### What is changed and how it works?
This PR refers to the design of Flink `Configuration` and unifies the behaviors of StreamX configuration initialization,  default value and automatic overlay from `application.yml`. 
The definition of configuration is similar to the following:
```scala
  val STREAMX_WORKSPACE_LOCAL: ConfigOption = ConfigOption(
    key = "streamx.workspace.local",
    defaultValue = "/streamx",
    classType = classOf[java.lang.String])

  val STREAMX_WORKSPACE_REMOTE: ConfigOption = ConfigOption(
    key = "streamx.workspace.remote",
    defaultValue = "/streamx",
    classType = classOf[java.lang.String])

  val STREAMX_HADOOP_USER_NAME: ConfigOption = ConfigOption(
    key = "streamx.hadoop-user-name",
    defaultValue = "hdfs",
    classType = classOf[java.lang.String])

  val DOCKER_IMAGE_NAMESPACE: ConfigOption = ConfigOption(
    key = "streamx.docker.register.image-namespace",
    defaultValue = "streamx",
    classType = classOf[java.lang.String],
    description = "namespace for docker image used in docker building env and target image register")

  val MAVEN_REMOTE_URL: ConfigOption = ConfigOption(
    key = "streamx.maven.remote-url",
    defaultValue = "https://repo1.maven.org/maven2/",
    classType = classOf[java.lang.String],
    description = "maven repository used for built-in compilation")

  val metricTrkTaskIntervalSec: ConfigOption = ConfigOption(
    key = "streamx.flink-k8s.tracking.polling-interval-sec.cluster-metric",
    defaultValue = 10L,
    classType = classOf[java.lang.Long],
    description = "interval seconds between two single flink-k8s metrics tracking task")

```
  
All configurations will automatically initialized from the spring configuration items of the same name. The override order of the configuration is `application.yaml -> default value`, We can use these value-safe configuration directly in our code:
```scala
// scala code
val mavenUrl: String = ConfigHub.get(MAVEN_REMOTE_URL)
val metricTrkInterval: Long = ConfigHub.get(metricTrkTaskIntervalSec)
// java code
String mavenUrl = ConfigHub.get(MAVEN_REMOTE_URL)
Long metricTrkInterval  = ConfigHub.get(metricTrkTaskIntervalSec)
```
or via key directly:
```scala
// scala code
val mavenUrl: String = ConfigHub.get("streamx.maven.remote-url")
// java code
String mavenUrl = ConfigHub.get("streamx.maven.remote-url")
```

<br>

### CODE REVIEW

**Comment by Al-assad** [comment]
@xinzhuxiansheng The issue #552 can be developed quickly based on the features provided by this PR.

**Review by wolfboys — COMMENTED** [review]
LGTM. ConfigHub or ConfigProvider, which name is better ?

File: streamx-common/src/main/scala/com/streamxhub/streamx/common/conf/ConfigConst.scala [hunk 1]
```diff
@@ -289,27 +285,20 @@ object ConfigConst {
...
-  val KEY_STREAMX_WORKSPACE_REMOTE = "streamx.workspace.remote"
-
-  val STREAMX_WORKSPACE_DEFAULT = "/streamx"
-
   /**
-   * maven repository used for built-in compilation
+   * flink config key
    */
-  val DEFAULT_MAVEN_REMOTE_URL = "https://repo1.maven.org/maven2/"
+  var KEY_FLINK_APPLICATION_ARGS = "$internal.application.program-args"
```
Comment (Al-assad) [inline streamx-common/src/main/scala/com/streamxhub/streamx/common/conf/ConfigConst.scala]: need to be modified to val

**Comment by Al-assad** [comment]
> LGTM. ConfigHub or ConfigProvider, which name is better ?

Both are similar, ConfigHub is easier to call with fewer words :)

**Comment by xinzhuxiansheng** [comment]
> @xinzhuxiansheng The issue #552 can be developed quickly based on the features provided by this PR.

@Al-assad  Do you suggest that I add docker-java client configuration to the "Al-assad:unified_conf" branch?

**Comment by xinzhuxiansheng** [comment]
@Al-assad   Do I need to wait until dev merges your code？then develop.

**Comment by Al-assad** [comment]
@xinzhuxiansheng  Nope, wait for this pr to merge to dev branch then develop based on the merged dev branch.

**Comment by xinzhuxiansheng** [comment]
> @xinzhuxiansheng Nope, wait for this pr to merge to dev then develop based on the merged dev.

Noted with thanks.

**Pull request merged**